elide-graphql/src/main/java/com/yahoo/elide/graphql/parser/GraphQLEntityProjectionMaker.java

```diff
@@ -0,0 +1,537 @@
+                .build();
+
+        // initialize the attribute container
+        entityProjection.setAttributes(new HashSet<>());
+        entityField.getArguments().forEach(argument -> addArgument(argument, entityProjection));
+
+        entityField.getSelectionSet().getSelections().forEach(selection -> addSelection(selection, entityProjection));
+
+        return entityProjection;
+    }
+
+    /**
+     * Add a graphQL {@link Selection} to an {@link EntityProjection}
+     *
+     * @param fieldSelection field/fragment to add
+     * @param parentProjection projection that has this field/fragment
+     */
+    private void addSelection(Selection fieldSelection, final EntityProjection parentProjection) {
+        if (fieldSelection instanceof FragmentSpread) {
+            addFragment((FragmentSpread) fieldSelection, parentProjection);
+        } else if (fieldSelection instanceof Field) {
+            if (EDGES.equals(((Field) fieldSelection).getName())
+                    || NODE.equals(((Field) fieldSelection).getName())) {
+                // if this graphql field is 'edges' or 'node', go one level deeper in the graphql document
+                ((Field) fieldSelection).getSelectionSet().getSelections().forEach(
+                        selection -> addSelection(selection, parentProjection));
+            } else {
+                addField((Field) fieldSelection, parentProjection);
+            }
+        } else {
+            throw new InvalidEntityBodyException(
+                    String.format("Unsupported selection type {%s}.", fieldSelection.getClass()));
+        }
+    }
+
+    /**
+     * Resolve a graphQL {@link FragmentSpread} into {@link Selection}s and add them to an {@link EntityProjection}
+     *
+     * @param fragment graphQL fragment
+     * @param parentProjection entity projection that contains this fragment
+     */
+    private void addFragment(FragmentSpread fragment, EntityProjection parentProjection) {
+        String fragmentName = fragment.getName();
+        if (!fragmentResolver.contains(fragmentName)) {
+            throw new InvalidEntityBodyException(String.format("Unknown fragment {%s}.", fragmentName));
+        }
+
+        FragmentDefinition fragmentDefinition = fragmentResolver.get(fragmentName);
+
+        // type name in type condition of the Fragment must match the entity projection type name
+        if (entityDictionary.getJsonAliasFor(parentProjection.getType())
+                .equals(fragmentDefinition.getTypeCondition().getName())) {
+            fragmentDefinition.getSelectionSet().getSelections()
+                    .forEach(selection -> addSelection(selection, parentProjection));
+        }
+    }
+
+    /**
+     * Add a new graphQL {@link Field} into an {@link EntityProjection}
+     *
+     * @param field graphQL field
+     * @param parentProjection entity projection that contains this field
+     */
+    private void addField(Field field, EntityProjection parentProjection) {
```

Comment:
Cyclomatic complexity and line length are quite large.  How did this pass checkstyles?  Let's break this function into more functions.

Reply:
👍